File: Array and String/reverse-vowel-of-a-string/Reverse Vowels of a String.go
func isVowel(c rune) bool {
    vowel := []rune{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
    for _, v := range vowel {
        if c == v {
            return true
        }
    }
    return false
}

func reverseVowels(s string) string {
    l, r := 0, len(s)-1
    runes := make([]rune, len(s))
    for idx, r := range s {
        runes[idx] = r
    }

    for l < r {
        if !isVowel(runes[r]) {
            r--
            continue
        }

        if !isVowel(runes[l]) {
            l++
            continue
        }

        runes[l], runes[r] = runes[r], runes[l]
        l++
        r--
    }

    return string(runes)
}

File: Array and String/reverse-vowel-of-a-string/Reverse Vowels of a String.py
#make empty array for vowel
#iteration for storing vowel from back to front
#iteration for changing vowel from front to back according to array

class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = ["a","i","u","e","o","A","I","U","E","O"]
        s_list = list(s)
        pointL, pointR = 0, len(s_list) - 1
        while pointL < pointR:
            if s_list[pointL] not in vowels:
                pointL += 1
                continue

            if s_list[pointR] not in vowels:
                pointR -= 1
                continue

            s_list[pointL], s_list[pointR] = s_list[pointR], s_list[pointL]
            pointL += 1
            pointR -= 1

        return "".join(s_list)
